sklearn/ensemble/_stacking.py

```diff
@@ -165,8 +165,11 @@ def fit(self, X, y, sample_weight=None):
         est_fitted_idx = 0
         for name_est, org_est in zip(names, all_estimators):
             if org_est != "drop":
-                self.named_estimators_[name_est] = self.estimators_[est_fitted_idx]
+                current_estimator = self.estimators_[est_fitted_idx]
+                self.named_estimators_[name_est] = current_estimator
                 est_fitted_idx += 1
+                if hasattr(current_estimator, "feature_names_in_"):
+                    self.feature_names_in_ = current_estimator.feature_names_in_
```

Comment:
This would set `self.feature_names_in_` always to the ones of the last estimator, i.e. `self.estimators_[-1].feature_names_in_`.
As all estimators get the same `X`, why not do it once?

Reply:
Because I was too lazy to write a function that checks that they features of different sub-estimators are consistent with one another, and that since they are all probably using `_validate_data` internally to set their `feature_names_in_` it's very likely that they will be consistent with one another in 99%+ of the case ;)

Reply:
> As all estimators get the same X, why not do it once?

Isn't the code shorter this way?

Reply:
Thanks for explaining. It's for sure not a bottleneck and, if necessary, could be improved in later PRs.